### PR TITLE
RabbitMQ: provide separate paremeters for user and password

### LINKED ISSUE #2513: Rabbitmq scaler: Allow providing separate paremeters for user and password
### Proposal

I would like to be able to provide username/password as separate parameters for the RabbitMQ scaler, in order to be able to get host from Environment variable and username/password from separate environment variables/secrets

### Use-Case

We are using single RabbitMQ deployment for multiple applications which have separate secrets containing their credentials. Since credentials are generated by helm templates we have no way to know what they will be in advance. Adding to this we are using podpreset to inject the environment variables containing the username and password, which means we cannot combine everything in 1 variable using the pod manifest, because the username/password variables are injected at the end of the variables array. Having the option to reference separate secret/environment variables from which username and password to be taken is crucial for such deployments.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dovics**
_Provide a description of what has been changed_

rabbitmq using separate username and password, supports obtaining from environment variables or obtaining from TriggerAuthentication, and remains compatible with previous usage.

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Tests have been added
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #2513

Relates to https://github.com/kedacore/keda-docs/pull/1472

### CODE REVIEW

File: tests/scalers/rabbitmq/rabbitmq_queue_http_auth/rabbitmq_queue_http_auth_test.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	deploymentName            = fmt.Sprintf("%s-deployment", testName)
+	triggerAuthenticationName = fmt.Sprintf("%s-ta", testName)
+	secretName                = fmt.Sprintf("%s-secret", testName)
+	scaledObjectName          = fmt.Sprintf("%s-so", testName)
+	queueName                 = "hello"
+	user                      = fmt.Sprintf("%s-user", testName)
+	password                  = fmt.Sprintf("%s-password", testName)
+	vhost                     = "/"
+	NoAuthConnectionString    = fmt.Sprintf("http://rabbitmq.%s.svc.cluster.local", rmqNamespace)
+	connectionString          = fmt.Sprintf("amqp://%s:%s@rabbitmq.%s.svc.cluster.local", user, password, rmqNamespace)
```
[the remark this answers is not in the document]
Comment (dovics) [reply tests/scalers/rabbitmq/rabbitmq_queue_http_auth/rabbitmq_queue_http_auth_test.go:38]: Should this be acceptable in an e2e test? I could use the following, but it's a bit verbose

```
func buildConnectionString(username, password, namespace string) string {
    u := &url.URL{
        Scheme: "amqp",
        Host:   fmt.Sprintf("rabbitmq.%s.svc.cluster.local", namespace),
    }
    if username != "" && password != "" {
        u.User = url.UserPassword(username, password)
    }
    return u.String()
}
```

File: tests/scalers/rabbitmq/rabbitmq_queue_amqp_auth/rabbitmq_queue_amqp_auth_test.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+	deploymentName            = fmt.Sprintf("%s-deployment", testName)
+	triggerAuthenticationName = fmt.Sprintf("%s-ta", testName)
+	secretName                = fmt.Sprintf("%s-secret", testName)
+	scaledObjectName          = fmt.Sprintf("%s-so", testName)
+	queueName                 = "hello"
+	user                      = fmt.Sprintf("%s-user", testName)
+	password                  = fmt.Sprintf("%s-password", testName)
+	vhost                     = "/"
+	NoAuthConnectionString    = fmt.Sprintf("amqp://rabbitmq.%s.svc.cluster.local", rmqNamespace)
+	connectionString          = fmt.Sprintf("amqp://%s:%s@rabbitmq.%s.svc.cluster.local", user, password, rmqNamespace)
```
[the remark this answers is not in the document]
Comment (JorTurFer) [reply tests/scalers/rabbitmq/rabbitmq_queue_amqp_auth/rabbitmq_queue_amqp_auth_test.go:38]: yeah, e2e test aren't a problem or they shouldn't be

**Review by JorTurFer — APPROVED** [review]
LGTM! Thanks for the improvement :)

**Comment by JorTurFer** [comment]
@kedacore/keda-core-contributors PTAL

**Comment by JorTurFer** [comment]
/run-e2e rabbit
**Update:** You can check the progress [here](https://github.com/kedacore/keda/actions/runs/11152114907)

**Review by zroubalik — APPROVED** [review]
LGTM, thanks!

**Review by Gallardot — APPROVED** [review]
LGTM!

**Review by wozniakjan — APPROVED** [review]
lgtm, thank you!

happy to merge this as is but just wanted to ask @dovics, are you willing to also refactor this scaler as outlined in https://github.com/kedacore/keda/issues/5797? Could be in a separate PR, I think this one packs good changes already.

**Pull request merged**